src/v/cloud_roles/apply_abs_oauth_credentials.cc

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "apply_abs_oauth_credentials.h"
+
+namespace cloud_roles {
+apply_abs_oauth_credentials::apply_abs_oauth_credentials(
+  abs_oauth_credentials const& credentials)
+  : _oauth_token{fmt::format("Bearer {}", credentials.oauth_token())} {}
+
+std::error_code apply_abs_oauth_credentials::add_auth(
+  http::client::request_header& header) const {
+    auto token = _oauth_token();
+    // x-ms-version requests a specific version for the abs api, the hardcoded
+    // value is the one we test
```

Comment:
> the hardcoded value is the one we test

what does this mean? is it relevant for this context? it seems the important comment is the one that is attached to `azure_storage_api_version`?